src/shared/seccomp-util.c

```diff
@@ -891,9 +891,13 @@ int seccomp_add_syscall_filter_item(scmp_filter_ctx *seccomp, const char *name,
                 r = seccomp_rule_add_exact(seccomp, action, id, 0);
                 if (r < 0) {
                         /* If the system call is not known on this architecture, then that's fine, let's ignore it */
-                        if (log_missing)
-                                log_debug_errno(r, "Failed to add rule for system call %s() / %d, ignoring: %m",
-                                                name, id);
+                        bool ignore = r == -EDOM;
+
+                        if (!ignore || log_missing)
+                                log_debug_errno(r, "Failed to add rule for system call %s() / %d%s: %m",
+                                                name, id, ignore ? ", ignoring" : "");
+                        if (!ignore)
+                                return r;
```

Comment:
uh, we initially had more precise error checking here, but i gradually removed this, as the error reporting by libseccomp was frickin unreliable, i.e. see commits like 448ac526a3fc95c4b4bfa24fe25ea95a88127c49 or 1c6af69b2d25af90e210bc21cc378fb4a6c96316 or ff217dc3afe95504e48aeb8d8ad7fb7f53ce9cb1

Has libseccomp been fixed since then? Are you sure you want to rely on that? If it has been been fixed, we really need to add a versioned dep onto the right version of libseccomp, so that this doesn't fall apart for some users...

I think if anything we should ignore all errors except for a few safe, known candidates, such as EPERM/EACCES or so...

Reply:
This doesn't change the logic added in 448ac52 or 1c6af69 or ff217dc. It effectively only checks calls to `seccomp_rule_add_exact`, which should return -EDOM (which we ignore), or stuff like -ENOMEM on oom and -EINVAL if the arguments are wrong (e.g. we pass an unknown action), which we want to know about. If you can see a case where the tightened check fails spuriously, I'll be happy to fix it. But I think we shouldn't just ignore all errors.

Also, note that the libseccomp in our CI is pretty old. And the one I have on my machine is pretty new. We should already cover a wide range of versions in testing.